File: Mcs-cs/Mcs/Readers/InputReader.cs
using Mcs.Models;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Mcs.Readers
{
    public class InputReader
    {
        public static Input Read(string inputFilePath)
        {
            using (StreamReader reader = new StreamReader(inputFilePath))
            {
                Graph g1 = ReadGraph(reader);
                Graph g2 = ReadGraph(reader);
                return new Input(g1, g2);
            };
        }

        private static Graph ReadGraph(StreamReader reader)
        {
            if (!int.TryParse(reader.ReadLine(), out int n))
            {
                throw new SystemException("Unable to read size of the graph!");
            }
            List<List<int>> adjacentMatrix = new List<List<int>>();
            for (int i = 0; i < n; i++)
            {
                adjacentMatrix.Add(new List<int>());
                string[] inputs = (reader.ReadLine() ?? "").Trim(' ').Split();
                if (inputs.Length != n)
                {
                    throw new SystemException("Row " + i + " of adjacency matrix has " + inputs.Length + " items instead of " + n);
                }
                for (int j = 0; j < n; j++)
                {
                    if (!int.TryParse(inputs[j], out int value))
                    {
                        throw new SystemException("Unable to read AdjacencyMatrix[" + i + "][" + j + "] item!");
                    }
                    adjacentMatrix[i].Add(value);
                }
            }
            return new Graph(n, adjacentMatrix);
        }
    }
}

File: mcs-java/src/main/java/anc/readers/InputReader.java
package anc.readers;

import java.io.File;
import java.io.IOException;
import java.util.Scanner;

import anc.models.Graph;
import anc.models.Input;

public class InputReader {

    public static Input read(String inputFilePath) {
        try (Scanner scanner = new Scanner(new File(inputFilePath))) {
            int n1 = scanner.nextInt();
            int[][] M1 = new int[n1][n1];
            for (int i = 0; i < n1; ++i)
                for (int j = 0; j < n1; ++j) {
                    M1[i][j] = scanner.nextInt();
                }
            Graph g1 = new Graph(n1, M1);

            int n2 = scanner.nextInt();
            int[][] M2 = new int[n2][n2];
            for (int i = 0; i < n2; ++i)
                for (int j = 0; j < n2; ++j) {
                    M2[i][j] = scanner.nextInt();
                }
            Graph g2 = new Graph(n2, M2);

            return new Input(g1, g2);
        } catch (IOException e) {
            throw new RuntimeException("Error with reading input file: " + e.getMessage());
        }
    }
}
